ocs_ci/ocs/bucket_utils.py

```diff
@@ -55,13 +55,18 @@ def craft_s3_command(cmd, mcg_obj=None, api=False, signed_request_creds=None):
             region = f"AWS_DEFAULT_REGION={mcg_obj.region} "
         else:
             region = ""
+        endpoint = (
+            mcg_obj.s3_endpoint
+            if config.ENV_DATA["platform"] in constants.HCI_PC_OR_MS_PLATFORM
+            else mcg_obj.s3_internal_endpoint
+        )
```

Comment:
Since This code is repeated in multiple places, I would suggest to create one separate method for capturing endpoint and call that method wherever required.

Reply:
I agree, please capture this under a method in the MCG class - something like `determine_endpoint`. We could leverage it in a potential future refactor where we only use `MCG::s3_endpoint` as a property.